src/background.ts

```diff
@@ -99,14 +99,46 @@ protocol.registerSchemesAsPrivileged([
 
 const isMac = process.platform === "darwin";
 
+function detectImageTypeFromBase64(data: string): string {
+  switch (data[0]) {
+    case "/":
+      return "image/svg+xml";
+    case "R":
+      return "image/gif";
+    case "i":
+      return "image/png";
+    case "D":
+      return "image/jpeg";
+    default:
+      return "";
+  }
+}
+
 const engineInfos: EngineInfo[] = (() => {
   const defaultEngineInfosEnv = process.env.DEFAULT_ENGINE_INFOS;
+  let engines: EngineInfo[] = [];
 
   if (defaultEngineInfosEnv) {
-    return JSON.parse(defaultEngineInfosEnv) as EngineInfo[];
+    engines = JSON.parse(defaultEngineInfosEnv) as EngineInfo[];
   }
 
-  return [];
+  return engines.map((engineInfo: EngineInfo) => {
+    if (!engineInfo.icon) return engineInfo;
+    let b64icon: string;
+    try {
+      b64icon = fs.readFileSync(path.resolve(appDirPath, engineInfo.icon), {
+        encoding: "base64",
+      });
+    } catch (e) {
+      log.error("Failed to read icon file: " + engineInfo.icon);
+      return engineInfo;
+    }
+    return {
+      ...engineInfo,
+      icon: `data:${detectImageTypeFromBase64(b64icon)};base64,${b64icon}`,
+      path: engineInfo.path && path.resolve(appDirPath, engineInfo.path),
```

Comment:
ここの意図としては、engineInfo.pathが空欄やundefinedだったらappDirPathを指定する、という感じでしょうか。
だとすると`path.resolve(appDirPath, engineInfo.path)`よりappDirPathをそのまま指定するほうがそれっぽいのかなと思いました。

Reply:
pathが無いときはそのままにする、みたいな感じです。

Reply:
なるほどです。だとするとこっちの方が直感的かなーと
```suggestion
      path: engineInfo.path && appDirPath,
```

Reply:
これの役割が、
・pathが存在しなかったら/空欄だったらそのまま
・存在したら絶対パスに加工
っていう感じなんですよね。
path.joinじゃないのはengineInfo.pathが絶対パスだったときの処理を入れてます。

Reply:
あ！なるほどです！（&&の挙動を||と勘違いしていました。。）

空欄のときだけ空欄、それ以外は絶対パスになる感じでしょうか。
`engineInfo.path && `をなくして、どんなときでも絶対パスになるようにすると扱いやすいのかなと思いました！

Reply:
.envとかで指定する時のために、一応pathは省略可能にしているのでこうなってしまうんですけれど、省略不可にします？

Reply:
省略可能でいいと思うのですが、省略したときはpathが`appDirPath`になったほうが扱いやすそう、という感じです！

Reply:
そうすると、エンジンがappDirPathのところで合っているのか、キーが存在しないのかわからなくなるんですよね。
そうなると「フォルダを開く」ボタンの非表示の処理がかなり面倒なことになりそうです。

Reply:
あ、なるほどです！！！！すごくよくわかりました！
undefinedのときはundefined、それ以外（空白含む）だと絶対パス、でどうでしょう。こんな感じとか。
```suggestion
      path: engineInfo.path == undefined && path.resolve(appDirPath, engineInfo.path),
```

Reply:
反映させましたー。